### PR TITLE
frontend/backend: modify transaction loading flow and add fiat value at time of transaction

### PR DESCRIPTION
**Pull request by Beerosagos**
/transactions endpoint modified to load only data displayed in transactionfrontend list

A new /transaction endpoint created to load all available data for specific transactionin order to populate the detailed transaction view in frontend

FiatConversion component modified to use backend-calculated fiat amount conversionswhen available

Added code to show fiat value at time of transaction in detailed transaction view

### CODE REVIEW

**Comment by Beerosagos** [comment]
I am not sure about how to handle localization, so I started adding locales for IT and EN. Is that ok?

**Comment by Beerosagos** [comment]
One more thing: I only tested with TBTC for now, but I think it would be better to test also in ETH, if possible.

**Comment by thisconnect** [comment]
add new keys to EN, I am not sure if IT will be imported

existing keys need to be changed in locize and then pulled in a different PR

File: frontends/web/src/components/transactions/transaction.tsx [hunk 1]
```diff
@@ -63,11 +64,14 @@ class Transaction extends Component<Props, State> {
...
-      editMode: !this.props.note,
-    });
+    accountApi.getTransaction(this.props.accountCode, this.props.internalID).then(transaction => {
+      this.setState({
+        transactionInfo: transaction,
+        transactionDialog: true,
+        newNote: this.props.note,
+        editMode: !this.props.note,
+      });
+    })
```
Comment (thisconnect) [inline frontends/web/src/components/transactions/transaction.tsx]: please add semicolon; (I should configure the linter that semicolons are required)

File: frontends/web/src/components/rates/rates.tsx [hunk 2]
```diff
@@ -121,6 +121,7 @@ export function formatCurrency(amount: number, fiat: Fiat): string {
 export interface AmountInterface {
     amount: string;
     unit: Coin;
+    conversions?: Conversions;
 }
```
Comment (thisconnect) [inline frontends/web/src/components/rates/rates.tsx]: There is already an `IAmount` interface with the same fields in `api/account.ts`, but conversions is not optional… I did a quick test and made it optional and TypeScript did not complain, maybe use the one from `api/account.ts` and make it optional? 

https://github.com/digitalbitbox/bitbox-wallet-app/blob/197c8b7d4113d71bf717d391a70eda3b5f9b4cac/frontends/web/src/api/account.ts#L135-L139

File: frontends/web/src/components/rates/rates.tsx [hunk 3]
```diff
@@ -148,8 +149,12 @@ function Conversion({
   }
   const coin = amount.unit;
   let formattedValue = '';
-  if (rates[coin]) {
-    formattedValue = formatCurrency(rates[coin][active] * Number(amount.amount), active);
+  if (amount.conversions) {
+    formattedValue = amount.conversions[active];
+  } else {
+    if (rates[coin]) {
+      formattedValue = formatCurrency(rates[coin][active] * Number(amount.amount), active);
```
Comment (thisconnect) [inline frontends/web/src/components/rates/rates.tsx:152]: I believe you mentioned that this is only used in the send view is that correct? maybe add a comment?

File: frontends/web/src/components/transactions/transaction.tsx [hunk 4]
```diff
@@ -111,15 +115,8 @@ class Transaction extends Component<Props, State> {
...
       type,
-      txID,
       amount,
-      fee,
       feeRatePerKb,
-      gas,
-      nonce,
-      vsize,
-      size,
-      weight,
```
Comment (thisconnect) [inline frontends/web/src/components/transactions/transaction.tsx:122]: nice 👍 , much cleaner with `transactionInfo`, just one note, those props were able to live update ( data passed  from `account.ts`) whereas now the data is loaded once, on `showDetails`. Most of the fields don't change but I believe some do, i.e. numConfirmations. I make a comment on that line...

File: frontends/web/src/components/transactions/transaction.tsx [hunk 5]
```diff
@@ -251,7 +249,7 @@ class Transaction extends Component<Props, State> {
               </div>
               <div className={style.detail}>
                 <label>{t('transaction.confirmation')}</label>
-                <p>{numConfirmations}</p>
+                <p>{transactionInfo.numConfirmations}</p>
```
Comment (thisconnect) [inline frontends/web/src/components/transactions/transaction.tsx]: If I am not missing something I believe `numConfirmations` did live update (data passed through the components from `account.tsx`). I would keep using `numConfirmations`  from props so it still can update once there is a new block.

**Review by thisconnect — CHANGES_REQUESTED** [review]
Nice @Beerosagos 👍 

Left some comments, I think there is an issue with live updating. I'll do some testing on master.

also squash the 2 commits into 1?

File: frontends/web/src/components/rates/rates.tsx (hunk 3 above)
Comment (Beerosagos) [reply frontends/web/src/components/rates/rates.tsx:152]: I made a deeper check. It seems to me that it is used in sending https://github.com/Beerosagos/bitbox-wallet-app/blob/07ee89f61397c26f34cd7877dacf6592c2acf7b3/frontends/web/src/routes/account/send/utxos.tsx#L108 but also in the account summary https://github.com/Beerosagos/bitbox-wallet-app/blob/07ee89f61397c26f34cd7877dacf6592c2acf7b3/frontends/web/src/routes/account/summary/accountssummary.tsx#L262
If I am correct, maybe we could make a dedicated issue for the account summary, to get fiat amount already converted by backend?

Comment (thisconnect) [reply frontends/web/src/components/rates/rates.tsx:152]: ack, yes a separate issue/PR makes sense.

Comment (Beerosagos) [reply frontends/web/src/components/rates/rates.tsx:152]: UPDATE: after further checks I think that the only point where the conversion is done by the frontend is in the account summary, where it computes the total fiat balance of several accounts of the same coin. 
https://github.com/Beerosagos/bitbox-wallet-app/blob/07ee89f61397c26f34cd7877dacf6592c2acf7b3/frontends/web/src/routes/account/summary/accountssummary.tsx#L262

Could be probably fixed with a call to `coins/convertToFiat` like in: https://github.com/Beerosagos/bitbox-wallet-app/blob/07ee89f61397c26f34cd7877dacf6592c2acf7b3/frontends/web/src/routes/account/send/send.tsx#L395

I'll make a dedicated issue.

File: frontends/web/src/components/transactions/transaction.tsx (hunk 1 above)
Comment (Beerosagos) [reply frontends/web/src/components/transactions/transaction.tsx]: Done!

File: frontends/web/src/components/rates/rates.tsx (hunk 2 above)
Comment (Beerosagos) [reply frontends/web/src/components/rates/rates.tsx]: Good point! making it optional doesn't seem to rise any issue. Done.

File: frontends/web/src/components/transactions/transaction.tsx (hunk 4 above)
Comment (Beerosagos) [reply frontends/web/src/components/transactions/transaction.tsx:122]: Done!

File: frontends/web/src/components/transactions/transaction.tsx (hunk 5 above)
Comment (Beerosagos) [reply frontends/web/src/components/transactions/transaction.tsx]: You are right, fixed it! Thx.

**Comment by Beerosagos** [comment]
Thanks for the review! Now should be better.

While testing again I noticed a small bug: when I create a new transaction, the historical fiat value won't be available until the next rates update, but it is not a field meant for live updating. This means that it won't be displayed until I open the transaction detail again. 
If this is acceptable (and my new commit is fine) I can squash the commits a make a new PR upstream, otherwise I can include that value to those loaded by the /transactions endpoint and have it updated too. 
wdyt?

**Comment by thisconnect** [comment]
so the options are:
- show current amount
- show placeholder "N/A" or dashes
- conditionally render the row only if the historical amount is available

I slightly prefer to only render the row if there is a historical value.

**Comment by Beerosagos** [comment]
> 
It currently works as in your second option: when the historical value is not available there are dashes instead. But should be easy to render the row only if the value is available.

**Comment by thisconnect** [comment]
@jadzeidan WDYT should be displayed if the historical value is not available right after a new TX was made (or if it happens for any other reason)?

**Comment by Beerosagos** [comment]
Pushed a new commit to hide historical fiat value for unconfirmed transactions.

**Comment by thisconnect** [comment]
LGTM, time to squash and open a PR on the main repo for marko to review?

**Comment by Beerosagos** [comment]
upstream PR created: https://github.com/digitalbitbox/bitbox-wallet-app/pull/1724

Closing this PR

**Pull request closed without merging**